File: AIM/AIM/Domain/UseCases/GetOnAirInfoUseCase.m
//
//  GetOnAirInfoUseCase.m
//  AIM
//
//  Created by Oskar Szydlowski on 04.03.2018.
//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
//

#import "GetOnAirInfoUseCase.h"

@implementation GetOnAirInfoUseCase

- (id) initWithGateway:(id<OnAirInfoGateway>)gateway completion:(OnAirInfoCompletion)completion {
    self = [super init];
    if (self != nil) {
        self.gateway = gateway;
        self.completion = completion;
    }
    return self;
}

- (void) execute {
    [self.gateway fetchOnAirInfo:self.completion];
}

@end

File: AIM/AIM/Domain/UseCases/GetOnAirInfoUseCase.swift
//
//  GetOnAirInfoUseCase.swift
//  AIM
//
//  Created by Oskar Szydlowski on 02.03.2018.
//  Copyright © 2018 Oskar Szydlowski. All rights reserved.
//

class GetOnAirInfoUseCase {
    
    private let gateway: OnAirInfoGateway
    private let completion: OnAirInfoCompletion
    
    init(gateway: OnAirInfoGateway, completion: @escaping OnAirInfoCompletion) {
        self.gateway = gateway
        self.completion = completion
    }
    
    func execute() {
        gateway.fetchOnAirInfo(completion: completion)
    }
}
